### ISSUE TITLE
GLPK_MI aborts with `glp_add_cols: ncs = 0; invalid number of columns` on a small integer linear program

### ISSUE THREAD

**Issue by mvernacc** [cvxpy/cvxpy#1101]
**Describe the bug**
On a small, feasible, integer linear program, the GLPK_MI solver fails with the following error:
```
glp_add_cols: ncs = 0; invalid number of columns
Error detected in file api/prob1.c at line 362
Aborted (core dumped)
```
whereas other solvers, such as SCIP, solve the ILP correctly.

**To Reproduce**
Here is a minimum working example: https://gist.github.com/mvernacc/86e344670b9a014214b501cf3fc6ac73

**Expected behavior**
I believe the correct solution for the ILP in the example is `x = [1, 1, 0, 0]` with an objective value of `2`.

If I run `problem.solve(solver='SCIP')` instead of `problem.solve(solver='GLPK_MI')` in the example, I get the expected behavior, which is for the script to output:

```
optimal

problem.value, should be 2
2.0

x.value, should be [1, 1, 0, 0]
[1. 1. 0. 0.]
```

**Version**
 - OS: Ubuntu 20.04
 - CVXPY Version: 1.1.3
- CVXOPT Version: 1.2.5

**Comment by SteveDiamond** [comment]
The issue is that the CVXOPT GLPK_MI interface can't handle problems with no equality constraints. Feel free to raise the issue with the CVXOPT developers. There's no easy fix from the CVXPY end.

**Comment by SteveDiamond** [comment]
I added an appropriate error message.

**Comment by SteveDiamond** [comment]
Ok it's not so simple as I thought. Reopening.

**Comment by mvernacc** [comment]
Steven,
Thanks for looking into this.

I just pulled the latest cvxpy from master, however my MWE still fails; it does not raise the ValueError you added in https://github.com/cvxgrp/cvxpy/commit/603e2666adfbb88ff964657d56d20eac9b3243bb.

I also tried adding a trivial equality constraint to the MWE - https://gist.github.com/mvernacc/3f7bd8c777c47f447c594a3e2d4fed6d - this still fails with the same error message:
```
glp_add_cols: ncs = 0; invalid number of columns
Error detected in file api/prob1.c at line 362
Aborted (core dumped)
```

**Comment by SteveDiamond** [comment]
@mvernacc yes, I was mistaken about the source of the error. GLPK_MI solves problems just fine without equality constraints. So I don't know why your problem is failing.

**Comment by mjpieters** [comment]
I’m running into the same issue, with equality constraints. I’m guessing that at this point this is either a cvxopt or GLMK bug. Has this been reported to either project yet?

**Comment by mjpieters** [comment]
The same issue has been reported on the GLMK list as a regression: https://lists.gnu.org/archive/html/bug-glpk/2020-07/msg00011.html

**Comment by mjpieters** [comment]
Why was this closed? You didn't include any information as to why this is resolved now.

cvxopt is still [building wheels on ubuntu-latest](https://github.com/cvxopt/cvxopt/blob/1.3.0/.github/workflows/linux_build.yml), which is [currently Ubuntu 20.04](https://github.com/actions/runner-images/blob/40e93383ae0d51fdc2b95e29d7ab7656b3f024d2/README.md) (and was at the time the latest wheels were built), so includes [GLPK 4.65](https://packages.ubuntu.com/focal/libglpk-dev), which has this bug. Only when wheels are being built against Ubuntu 21.10 or newer (GH has 22.04 available) will those wheels include [GLPK 5.0](https://packages.ubuntu.com/impish/libglpk-dev).

The conda forge builds also [build against GLPK 4.65](https://github.com/conda-forge/cvxopt-feedstock/tree/main/.ci_support); I don't know if those can be upgraded to GLPK 5, the conda forge build process is quite opaque to me.

If you build cvxopt from source on a system with 5.0 this *may* be resolved (I haven't tested against 5.0), but anyone relying on the wheels published on pypi or on the conda forge builds, will still see this problem.

**Comment by mjpieters** [comment]
Clearly, I've missed something somewhere, because running the linked gist shows this is indeed solved with the latest cvxopt / cvxpy versions. Perhaps it's not the same regression I linked to on the glpk  list.

**Comment by SteveDiamond** [comment]
Sorry, this was closed by accident.
